cmake/CMakeLists.txt

```diff
@@ -891,17 +891,26 @@ if (onnxruntime_USE_CUDA)
   endif()
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${ONNXRUNTIME_CUDA_LIBRARIES})
 
-  # the following compute capabilities are deprecated in CUDA 11 Toolkit
-  if (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11)
-    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_30,code=sm_30") # K series
-    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_50,code=sm_50") # M series
-  endif()
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_52,code=sm_52") # M60
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_60,code=sm_60") # P series
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_70,code=sm_70") # V series
-  if (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11)
-    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_80,code=sm_80") # A series
+  if(${CMAKE_LIBRARY_ARCHITECTURE} STREQUAL "aarch64-linux-gnu")
```

Comment:
this results in error. see https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=189625&view=logs&j=5b022bb4-70a7-5401-8766-a8a7802c7150&t=fc69e05d-8264-53f4-d9bf-c0376ddf5453&l=89

i think it might need to be in quotes, like "${CMAKE_LIBRARY_ARCHITECTURE}"

Reply:
if(CMAKE_LIBRARY_ARCHITECTURE STREQUAL "aarch64-linux-gnu") -- this one should work

---
In reply to: [462216068](https://github.com/microsoft/onnxruntime/pull/4637#discussion_r462216068) [](ancestors = 462216068)

Reply:
Please update the instruction in the BUILD.md since the manual steps it mentioned are not required with this change.
[https://github.com/microsoft/onnxruntime/blob/master/BUILD.md#TensorRT](https://github.com/microsoft/onnxruntime/blob/master/BUILD.md#TensorRT)
Jetson TX1/TX2/Nano (ARM64 Builds)

---
In reply to: [462497083](https://github.com/microsoft/onnxruntime/pull/4637#discussion_r462497083) [](ancestors = 462497083,462216068)